app/api/v1/endpoints/admin.py

```
@@ -228,7 +240,197 @@ async def configure_system(
+    except Exception as e:
+        logger.error("Failed to update credit pricing", error=str(e))
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to update pricing: {str(e)}"
+        )
+
+
+@router.put("/strategy-pricing")
+async def update_strategy_pricing(
+    request: StrategyPricingRequest,
+    current_user: User = Depends(require_role(UserRole.ADMIN))
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)